docs/developers.md

```diff
@@ -74,3 +74,61 @@ Read more:
 
 - Effective Go [introduces formatting](https://golang.org/doc/effective_go.html#formatting).
 - The Go Wiki has a great article on [formatting](https://github.com/golang/go/wiki/CodeReviewComments).
+
+### Unit Tests
+
+Unit tests may be run locally via `make test`.
+
+### End-to-end Tests
+
+End-to-end tests for the DCOS, Kubernetes and OpenShift orchestrators may be run
+via `make test-{dcos,kubernetes,openshift}`.  The test process can optionally
+deploy and tear down a cluster as part of the test (this is enabled by default).
+You'll need access to an Azure subscription, as well as at least the following
+environment variables to be set:
+
+* `ACCOUNT_SUBSCRIPTION_ID`: Azure subscription UUID
+* `ACCOUNT_TENANT_ID`: Azure tenant UUID
+* `ACCOUNT_USER_CLIENT_ID`: Azure client ID
+* `ACCOUNT_USER_CLIENT_SECRET`: Azure client secret
+
+* `PROVISIONER_ACCOUNT_SUBSCRIPTION_ID`: same as ACCOUNT_SUBSCRIPTION_ID
```

Comment:
I don't remember setting these when I was testing, are you sure they are required?

Reply:
Ah, I see, it's an error message problem - if you set FOO_ID, ACCOUNT_FOO_ID defaults to that.  But if you don't, the error message indicates to set ACCOUNT_FOO_ID...  Will update.